chainer/functions/normalization/decorrelated_batch_normalization.py

```diff
@@ -0,0 +1,305 @@
+from chainer import backend
+from chainer import function_node
+from chainer.utils import argument
+from chainer.utils import type_check
+
+
+def _calc_axis_and_m(x_shape, batch_size, groups):
+    m = batch_size * groups
+    spatial_ndim = len(x_shape) - 2
+    spatial_axis = tuple(range(2, 2 + spatial_ndim))
+    for i in spatial_axis:
+        m *= x_shape[i]
+    return spatial_axis, m
+
+
+class DecorrelatedBatchNormalizationFunction(function_node.FunctionNode):
+
+    def __init__(self, groups=16, eps=2e-5, mean=None, projection=None,
+                 decay=0.9):
+        self.groups = groups
+
+        self.running_mean = mean
+        self.running_projection = projection
+
+        self.eps = eps
+        self.decay = decay
+        self.axis = None
+
+    def check_type_forward(self, in_types):
+        type_check.expect(in_types.size() == 1)
+        x_type = in_types[0]
+        type_check.expect(
+            x_type.dtype.kind == 'f',
+            x_type.shape[1] % self.groups == 0,
+        )
+        type_check.expect(
+            x_type.ndim >= 2,
+        )
+
+    def forward(self, inputs):
+        self.retain_inputs(())
+        x = inputs[0]
+        xp = backend.get_array_module(x)
+        x_shape = x.shape
+        b, c = x_shape[:2]
+        g = self.groups
+        C = c // g
+        spatial_axis, m = _calc_axis_and_m(x_shape, b, g)
+
+        if self.running_mean is None:
+            self.running_mean = xp.zeros(C, dtype=x.dtype)
+            self.running_projection = xp.eye(C, dtype=x.dtype)
+
+        if g > 1:
+            x = x.reshape((b * g, C) + x.shape[2:])
+        x_hat = x.transpose((1, 0) + spatial_axis).reshape(C, -1)
+
+        mean = x_hat.mean(axis=1)
+        x_hat = x_hat - mean[:, None]
+        self.eps = x.dtype.type(self.eps)
+
+        identity = self.eps * xp.eye(C, dtype=x.dtype)
```

Comment:
A nitpick but I find this variable name slightly misleading. I assume it's there for numerical stability, how about simply `eps` ?